test/controllers/update/update.statement.validation.errors.controller.spec.ts

```diff
@@ -177,6 +178,72 @@ describe("Update statement validation errors controller", () => {
+
+      expect(resp.status).toEqual(200);
+
+      expect(resp.text).toContain('The statements you&#39;ve chosen do not match the information provided in this update');
+      expect(resp.text).toContain(UPDATE_REGISTRABLE_BENEFICIAL_OWNER_URL);
+      expect(resp.text).toContain('Potato - OE991992');
+      expect(resp.text).toContain('There is at least one active managing officer.');
+      expect(resp.text).toContain('All beneficial owners have been identified and I can provide all the required information');
+      expect(resp.text).toContain('The entity has no reasonable cause to believe that anyone has become or ceased to be a registrable beneficial owner during the update period');
+
+      expect(resp.text).not.toContain(UPDATE_NO_CHANGE_REGISTRABLE_BENEFICIAL_OWNER_URL);
+      expect(resp.text).not.toContain('value="statement-resolution-change-information" checked');
+      expect(resp.text).not.toContain('value="statement-resolution-change-statement" checked');
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+    });
+
+    test(`in a no change journey, renders the Update statement validation errors page when there is at least one active managing officer`, async () => {
+      mockHasValidStatements.mockImplementation((req: Request, __: Response, next: NextFunction) => {
+        req['statementErrorList'] = ['There is at least one active managing officer.'];
+        next();
+      });
+      mockGetApplicationData.mockReturnValue({
+        entity_name: 'Potato',
+        entity_number: 'OE991992',
+        beneficial_owners_statement: 'ALL_IDENTIFIED_ALL_DETAILS',
+        managing_officers_individual: [ UPDATE_MANAGING_OFFICER_OBJECT_MOCK ],
+        update: {
+          no_change: true,
+          registrable_beneficial_owner: 0,
+        },
+      });
+
+      const resp = await request(app).get(UPDATE_STATEMENT_VALIDATION_ERRORS_URL);
+
+      expect(resp.status).toEqual(200);
+
+      expect(resp.text).toContain('The statements you&#39;ve chosen do not match the information provided in this update');
+      expect(resp.text).toContain(UPDATE_NO_CHANGE_REGISTRABLE_BENEFICIAL_OWNER_URL);
+      expect(resp.text).toContain('Potato - OE991992');
+      expect(resp.text).toContain('There is at least one active managing officer.');
+      expect(resp.text).toContain('All beneficial owners have been identified and I can provide all the required information');
+      expect(resp.text).toContain('The entity has no reasonable cause to believe that anyone has become or ceased to be a registrable beneficial owner during the update period');
+
+      expect(resp.text).not.toContain(UPDATE_REGISTRABLE_BENEFICIAL_OWNER_URL);
+      expect(resp.text).not.toContain('value="statement-resolution-change-information" checked');
+      expect(resp.text).not.toContain('value="statement-resolution-change-statement" checked');
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+    });
+
```

Comment:
this test file keeps growing and it is a lot of repeated checks - are these tests really necessary compared to ones that are already there? I don't think we need to have separate tests in here for every scenario because your tests in _test/middleware/statement.validation.middleware.spec.ts_ cover all the error scenarios.

You could just add your error message into `req['statementErrorList']` of an existing test but even then not sure what value you're getting out of that.

Reply:
yeah sweet go for it 💯 ! Don't think we'll lose anything without it